ui/components/MesheryPatternServiceForm.js

```diff
@@ -0,0 +1,206 @@
+// @ts-check
+import React from "react";
+import { Tab, Tabs, AppBar, Typography, IconButton } from "@material-ui/core";
+import { Close, Delete } from "@material-ui/icons";
+import PatternService from "./MesheryMeshInterface/PatternService"
+import useStateCB from "../utils/hooks/useStateCB";
+import PascalCaseToKebab from "../utils/PascalCaseToKebab";
+
+function TabPanel(props) {
+  const {
+    children, value, index, ...other
+  } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Typography>{children}</Typography>
+      )}
+    </div>
+  );
+}
+
+function a11yProps(index) {
+  return {
+    id : `simple-tab-${index}`,
+    "aria-controls" : `simple-tabpanel-${index}`,
+  };
+}
+
+function getPatternAttributeName(jsonSchema) {
```

Comment:
I saw this function on the other file, can we make it reusable instead of this duplicate definitions?

Reply:
@bariqhibat that is a great suggestion. Yes, we need to avoid repetition between patterns, application, and filters. Will you `gh pr checkout 4085` and assist? // @Chadha93 @sudo-NithishKarthik

Reply:
Actually, it is intended.
 80% of the file content inside `MesheryPatternServiceForm.js` is same as `MesheryMeshInterface/PatternServiceForm.js`.
 But this is done on purpose, since earlier I was actually sharing all the code with the `MesheryMeshInterface`, so both of the components are pre-mature and they are under rapid development. So whenever there is a change done thinking to make` MeshInterface` better, there is a side effect on the `Patterns page`. So earlier we had seen such conflicts, like the rjsf form styling issues inside patterns, the accordion expansion , and the syncup of contents. 
 
 That's why just tried to extract out the functionalities we can, and once both components become mature we can try and merge the two codes.
 What do you say? @bariqhibat @leecalcote

Reply:
I see, if both configurations are independent to each other, then it makes sense to separate them. And eventually both need to be merged. Could you see into it once they're at that point? @Abhishek-kumar09

Reply:
yeah, for sure :smile:

Reply:
Thank you, gents. Let's proceed according to your plan.